File: digittostr.py
def checkNum(d):
    num = 0
    
    if d <= 0:
        print("Invalid Low Number")
    elif d >= 1 and d < 10:
        num = 1
    elif d >= 10 and d < 100:
        num = 2
    elif d >= 100 and d < 1000:
        num = 3
    elif d >= 1000 and d < 10000:
        num = 4
    else:
        print("Invalid High Number")
        
    return num

def oneDigit(d):
    word = ""
    
    if(d == 1):
        word = "one"
    elif(d == 2):
        word = "two"
    elif(d == 3):
        word = "three"
    elif(d == 4):
        word = "four"
    elif(d == 5):
        word = "five"
    elif(d == 6):
        word = "six"
    elif(d == 7):
        word = "seven"
    elif(d == 8):
        word = "eight"
    elif(d == 9):
        word = "nine"
        
    return word

def twoDigits(d):
    if(d >= 10 and d < 20):
        if(d == 10):
            return "ten"
        elif(d == 11):
            return "eleven"
        elif(d == 12):
            return "twelve"
        elif(d == 13):
            return "thirteen"
        elif(d == 15):
            return "fifteen"
        else:
            teen = d % 10
            return oneDigit(teen) + "teen"
    else:
        first = d / 10
        second = d % 10
        word = ""
        
        if(first == 2):
            word = "twenty"
        elif(first == 3):
            word = "thirty"
        elif(first == 4):
            word = "forty"
        elif(first == 5):
            word = "fifty"
        elif(first == 6):
            word = "sixty"
        elif(first == 7):
            word = "seventy"
        elif(first == 8):
            word = "eighty"
        elif(first == 9):
            word = "ninety"
        
        if second != 0:
            word += oneDigit(second)
        return word

def threeDigits(d):
    first = d / 100
    rem = d % 100
    word = ""
    
    word = oneDigit(first) + " hundred "
    
    if(rem >= 10):
        word = word + twoDigits(rem)
    else:
        if(rem != 0):
            word = word + "and " + oneDigit(rem)
            
    return word

def fourDigits(d):
    first = d / 1000
    rem = d % 1000
    word = ""
    
    word  = oneDigit(first) + " thousand"
    if rem >= 100:
            word = word + " " + threeDigits(rem);
    elif rem >= 10:
            word = word + " " + twoDigits(rem);
    else:
        if rem != 0:
            word = word + " and " + oneDigit(rem); 
        
    return word;


def digitsF(n, d):
    word = ""
    
        
    return word

number = False
digits = 0
num = 0

while number == False:
    print("Please enter a number between 1 and 9999:")
    digits = int(input())
    num = checkNum(digits)
    if(num != 0):
        number = True

if(num == 1):
    word = oneDigit(digits)
elif(num == 2):
    word = twoDigits(digits)
elif(num == 3):
    word = threeDigits(digits)
else:
    word = fourDigits(digits)

print(word)

File: digit_to_str.R
num <- as.numeric(readline("Enter number:"))
n <- check(num)
switch(n,
       1 = print(one(num)),
       2 = print(two(num)),
       3 = print(three(num)),
       4 = print(four(num)),
       "unable to use program to convert num into words")

check <- function(num){
  if(num > 0 & num < 10){return (1)}
  else if(num >= 10 & num < 100){return (2)}
  else if(num >= 100 & num < 1000){return (3)}
  else if(num >= 1000 & num < 10000){return (4)}
  else {
    print("Invalid number entered")
    return (0)
  }
}
one <- function(num){
  dstr = ""
  switch(num,
         1 = {dstr <- "one"},
         2 = {dstr <- "two"},
         3 = {dstr <- "three"},
         4 = {dstr <- "four"},
         5 = {dstr <- "five"},
         6 = {dstr <- "six"},
         7 = {dstr <- "seven"},
         8 = {dstr <- "eight"}, 
         9 = {dstr <- "nine"},
         "invalid")
  return (dstr)
}
two <- function(num){
  dstr = ""
  if(num > 20){
    switch(num,
           10 = {dstr <- "ten"},
           11 = {dstr <- "eleven"},
           12 = {dstr <- "twelve"},
           13 = {dstr <- "thirteen"},
           14 = {dstr <- "fourteen"},
           15 = {dstr <- "fifteen"},
           16 = {dstr <- "sixteen"},
           17 = {dstr <- "seventeen"},
           18 = {dstr <- "eighteen"},
           19 = {dstr <- "nineteen"},
           "invalid")
  }
  else{
    nv <- as.numeric(strsplit(as.character(num), "")[[1]])
    switch(nv[c(1)],
           2 = {dstr <- "twenty"},
           3 = {dstr <- "thirty"}, 
           4 = {dstr <- "fourty"},
           5 = {dstr <- "fifty"},
           6 = {dstr <- "sixty"},
           7 = {dstr <- "seventy"},
           8 = {dstr <- "eighty"},
           9 = {dstr <- "ninety"},
           "invalid")
    if(nv[2] != 0){dstr <- dstr + " " + one(nv[2])}
  }
  return (dstr)
}
three<-function(num){
  nv <- as.numeric(strsplit(as.character(num), "")[[1]])
  dstr = ""
  if(nv[2] == 0 & nv[3]){dstr <- one(nv[1]) + " hundred"}
  else if(nv[2] == 0){dstr <- one(nv[1]) + " hundred and " + one(nv[3])}
  else {dstr <- one(nv[1]) + " hundred and " + two(merge(nv, 3))}
  
  return (dstr)
}
four<-function(num){
  nv <- as.numeric(strsplit(as.character(num), "")[[1]])
  dstr <- ""
  if(nv[2] == 0 & nv[3] == 0 & nv[4] == 0){dstr <- one(nv[1]) + " thousand"}
  else if(nv[2] == 0 & nv[3] == 0){dstr <- one(nv[1]) + " thousand and " + one(nv[4])}
  else if(nv[2] == 0){dstr <- one(nv[1]) + " thousand and " + two(merge(nv, 3))}
  else {dstr <- one(nv[1]) + " thousand " + three(merge(nv, 4))}
  
  return(dstr)
}
merge<-function(nv, len){
  nums<- as.character(nv)
  if(len == 3){num <- nums[2] + nums[3]}
  else if(len == 4){num <- nums[2] + nums[3] + nums[4]}
  return(as.numeric(num))
}